src/routes/scope-selection/ScopeSelection.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useMemo } from 'react'
+import styled, { useTheme } from 'styled-components/native'
+
+import DisciplineListItem from '../../components/DisciplineListItem'
+import SearchBar from '../../components/SearchBar'
+import ServerResponseHandler from '../../components/ServerResponseHandler'
+import { ContentTextBold, ContentTextLight } from '../../components/text/Content'
+import { Discipline } from '../../constants/endpoints'
+import { formatDiscipline } from '../../hooks/helpers'
+import { useLoadAllDisciplines } from '../../hooks/useLoadAllDisciplines'
+import { useLoadDisciplines } from '../../hooks/useLoadDisciplines'
+import { pushSelectedProfession } from '../../services/AsyncStorage'
+import { getLabels, searchProfessions, splitTextBySearchString } from '../../services/helpers'
+
+const SearchContainer = styled.View`
+  margin: ${props => props.theme.spacings.sm};
+`
+
+const ScopeContainer = styled.View`
+  margin: 0 ${props => props.theme.spacings.sm};
+  padding: ${props => props.theme.spacings.xs} 0;
+  background-color: ${props => props.theme.colors.background};
+`
+
+const StyledPressable = styled.Pressable`
+  padding: ${props => props.theme.spacings.sm};
+  border-bottom: 1px solid ${props => props.theme.colors.disabled};
+`
+
+const DisciplineContainer = styled.View`
+  margin: 0 ${props => props.theme.spacings.sm};
+`
+
+const HighlightContainer = styled.Text<{ disabled: boolean }>`
+  flex-direction: row;
+  color: ${props => (props.disabled ? props.theme.colors.disabled : props.theme.colors.black)};
+`
+
+const highlightText = (textArray: [string] | [string, string, string], disabled: boolean): JSX.Element => (
+  <HighlightContainer disabled={disabled}>
+    <ContentTextLight>{textArray[0]}</ContentTextLight>
+    <ContentTextBold>{textArray[1]}</ContentTextBold>
+    <ContentTextLight>{textArray[2]}</ContentTextLight>
+  </HighlightContainer>
+)
+
+type ScopeSelectionProps = {
+  queryTerm: string
+  setQueryTerm: (newString: string) => void
+  showSearchResults: boolean
+  setShowSearchResults: (shouldShow: boolean) => void
+  navigateToDiscipline: (item: Discipline) => void
+  navigateToManageSelection: () => void
+  selectedProfessions: number[] | null
+}
+
+const ScopeSelection = ({
+  queryTerm,
+  setQueryTerm,
+  showSearchResults,
+  setShowSearchResults,
+  navigateToDiscipline,
+  navigateToManageSelection,
+  selectedProfessions,
+}: ScopeSelectionProps): JSX.Element => {
+  const { data: disciplines, error, loading, refresh } = useLoadDisciplines({ parent: null })
+  const allProfessions = useLoadAllDisciplines()
+    .data?.filter(discipline => discipline.total_discipline_children === 0)
+    .map(discipline => formatDiscipline(discipline, {}))
```

Comment:
:question: Isn't the result of these exactly the same if `parent: null` is passed (except the `filter`)? AFAICS `useLoadAllDisciplines` queries the same url for `parent: null`. Therefore this could be unified. Not 100% sure though as the logic in `useLoadDisciplines` isn't super straight forward.

Reply:
It's not exactly the same. In the case of  `useLoadDisciplinces` with`parent: null`, we call the endpoint `/disciplines_by_level` but in the case of `useLoadAllDisciplines`, we call the endpoint `/disciplines`.
I could also put this into `useLoadDisciplines` as another special case but it really is already a pretty hard to understand file. What do you think?

Reply:
Ahh sorry, I guess I missed that. Thanks for the explanation, nvm :)